### PR TITLE
Another attempt to make VPC cleanup working

### PR DESCRIPTION
**Pull request by asmorodskyi**
Another attempt to make VPC cleanup working

1. Add missing check for pcw_ignore flag !
2. re-order deletion of dependent entities
3. remove elastic IP deletion ( looks like it is not needed )

### CODE REVIEW

File: ocw/lib/EC2.py [hunk 1]
```diff
@@ -64,6 +64,14 @@ def ec2_client(self, region: str) -> "boto3.session.Session.client":
...
         return self.__ec2_client[region]
 
+    def eks_client(self, region: str) -> "boto3.session.Session.client":
+        if region not in self.__eks_client:
+            self.__eks_client[region] = boto3.client('eks',
+                                                     aws_access_key_id=self.auth_json['access_key'],
+                                                     aws_secret_access_key=self.auth_json['secret_key'],
+                                                     region_name=region)
+        return self.__eks_client[region]
+
```
Comment (pdostal) [inline ocw/lib/EC2.py:74]: Is this needed?

File: ocw/lib/EC2.py [hunk 2]
```diff
@@ -190,27 +198,16 @@ def cleanup_all(self) -> None:
         if PCWConfig.getBoolean('cleanup/vpc_cleanup', self._namespace):
             self.cleanup_vpcs()
 
-    def delete_elastic_ips(self, region):
```
Comment (pdostal) [inline ocw/lib/EC2.py:193]: Is it causing issues?

File: ocw/lib/EC2.py [hunk 3]
```diff
@@ -190,27 +198,16 @@ def cleanup_all(self) -> None:
...
-            self.delete_vpc_endpoints(region, vpc_id)
+            self.delete_routing_tables(region, vpc_id)
             self.delete_security_groups(vpc)
-            self.delete_vpc_peering_connections(region, vpc_id)
             self.delete_network_acls(vpc)
             self.delete_vpc_subnets(vpc)
+            self.delete_internet_gw(vpc)
+            self.delete_vpc_endpoints(region, vpc_id)
+            self.delete_vpc_peering_connections(region, vpc_id)
             if self.dry_run:
```
Comment (pdostal) [inline ocw/lib/EC2.py:211]: Why is this here?

**Review by ricardobranco777 — APPROVED** [review]
LGTM

File: ocw/lib/EC2.py (hunk 3 above)
Comment (asmorodskyi) [reply ocw/lib/EC2.py:211]: because all functions which modify something on the cloud is guarded by dry_run variable so you have single switcher which makes PCW 100% harmless , which is very useful when we have some bug in code and PCW attempt to delete something what it should not . Or when new code is written and I have only "prod" environment to verify it

File: ocw/lib/EC2.py (hunk 2 above)
Comment (asmorodskyi) [reply ocw/lib/EC2.py:193]: with VPC deletion I never can say for 100% anything. The problem that VPC has plenty of different entities under control ( subnets , routing tables , routes , security groups , IPs , gateways and some others :) )  all they have some very complex relation between each other which I didn't manage fully realize. And if you trying to delete them in wrong order you will get exception. All changes which you see here is result of several days reading internet trigger different algorithms of deletion getting some exceptions and starting over again . 
So to answer you direct question **I don't know** but my gut feeling told me to drop it :D

File: tests/test_eks.py [hunk 4]
```diff
@@ -94,18 +94,24 @@ def __init__(self, name, age):
...
+
     monkeypatch.setattr(PCWConfig, 'get_feature_property', mock_get_feature_property)
     monkeypatch.setattr(Provider, 'read_auth_json',
                         lambda *args, **kwargs: {'access_key': 'key', 'secret_key': 'secret'})
-    monkeypatch.setattr(EKS, 'list_regions', lambda self: ['region1'])
-    monkeypatch.setattr(Provider, 'cmd_exec', lambda *args, **kwargs: MockedSubprocessReturn(0))
+    monkeypatch.setattr(Provider, 'cmd_exec', mocked_cmd_exec)
     monkeypatch.setattr(EKS, 'aws_dir', lambda self: '/tmp')
 
     # monkeypatch.setattr(EC2, 'check_credentials', lambda *args, **kwargs: True)
```
Comment (ilausuch) [inline tests/test_eks.py:117]: I think this is a leftover, we could remove this

**Review by ilausuch — APPROVED** [review]
only a small change, but the code is ok. Great job!

File: ocw/lib/EC2.py (hunk 2 above)
Comment (pdostal) [reply ocw/lib/EC2.py:193]: I see, OK :+1:

File: ocw/lib/EC2.py (hunk 3 above)
Comment (pdostal) [reply ocw/lib/EC2.py:160]: But why is it called only after the other functions?

Comment (asmorodskyi) [reply ocw/lib/EC2.py:160]: first we deleting all related resources and than we deleting vpc itself . 
if you asking why `if self.dry_run` not wrapping all function calls but only  exact call for deleting vpc - it is because each function internally has their own `if self.dry_run` checks . This way we doing everything what is possible w/o actually deleting resources , so when you have `dry_run = true ` you are closer to `dry_run = false` because you not skipping any querying calls but only deleting calls

Comment (pdostal) [reply ocw/lib/EC2.py:160]: I got it now. Thank you
